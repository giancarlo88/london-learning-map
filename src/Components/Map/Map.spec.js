import React from 'react'
import { shallow } from 'enzyme'

import MapComponent from './Map'

describe(MapComponent.name, () => {
  const defaultProps = {
    data: [],
    handleClick: () => {}
  }
  describe('@renders', () => {
    it('should render', () => {
      const wrapper = shallow(<MapComponent {...defaultProps} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('should render a <Marker /> for each entry in the data prop', () => {
      const data = [
        {
          _id: 1,
          index: 0,
          'x-cord': 1234,
          'y-cord': 2345
        }
      ]
      const props = {
        ...defaultProps,
        data
      }
      const wrapper = shallow(<MapComponent {...props} />)
      expect(wrapper.find('Marker').length).toBe(1)
    })
  })
})
