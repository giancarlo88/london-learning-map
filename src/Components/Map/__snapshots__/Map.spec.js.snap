// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapComponent @renders should render 1`] = `
<styled.div>
  <button />
  <GoogleMap
    bootstrapURLKeys={
      Object {
        "key": "AIzaSyBxejpla5DMrXFvEidgm90BJ-RsOs30KhE",
      }
    }
    debounced={true}
    defaultCenter={
      Object {
        "lat": 51.5199975,
        "lng": -0.1003068,
      }
    }
    defaultZoom={13}
    distanceToMouse={[Function]}
    googleMapLoader={[Function]}
    heatmap={Object {}}
    heatmapLibrary={false}
    hoverDistance={30}
    layerTypes={Array []}
    options={[Function]}
    style={
      Object {
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    yesIWantToUseGoogleMapApiInternals={false}
  >
    <Marker
      handleClick={[Function]}
      index={1}
      isSelected={false}
      key="1"
      lat={1234}
      lng={2345}
    />
  </GoogleMap>
</styled.div>
`;
